src/APIBasedAuth.ts

```diff
@@ -0,0 +1,192 @@
+import axios, {
+  AxiosError,
+  AxiosInstance,
+  AxiosRequestConfig,
+  AxiosResponse
+} from 'axios';
+
+/* istanbul ignore next */
```

Comment:
Can we remove these ignore statements before merging?

Reply:
I can, just need to figure out more tests then, it's just ignoring for coverage thresholds right now